### PR TITLE
Standardize secrets detection to use script-based approach

### LINKED ISSUE #145: Standardize secrets detection to use script-based approach
## Summary

Update the secrets detection setup in `template-repo-python` to use the standardized script-based approach adopted in other PDS Python repositories.

## 📋 Tasks

- [x] Add `scripts/detect_secrets_baseline.sh` — centralizes all scan arguments, per-repo exclude handling via `.detect-secrets-ignore`, and audit/check logic
- [x] Add `.detect-secrets-ignore` — per-repo file exclusion patterns (supplements global excludes in script)
- [x] Update `.pre-commit-config.yaml` — replace inline `slim-detect-secrets` remote hook with local `scripts/detect_secrets_baseline.sh` hook
- [x] Update `.github/workflows/secrets-detection.yaml` — install `detect-secrets~=1.5.0` from PyPI and delegate to script; fail fast if `.secrets.baseline` is missing
- [x] Update `setup.cfg` — add `detect-secrets~=1.5.0` to `[options.extras_require] dev` so it installs with `pip install --editable '.[dev]'`
- [x] Update `README.md` — replace manual `detect-secrets scan` block with script commands
- [x] Update `CLAUDE.md` — add secrets detection section with script usage

## 🎯 Motivation

The previous approach used a pinned commit hash from `slim-detect-secrets` with a long inline `--exclude-files` argument list duplicated across `.pre-commit-config.yaml` and the GitHub Actions workflow. The new approach:

- Uses `detect-secrets~=1.5.0` from standard PyPI (no pinned upstream fork commit)
- Keeps exclude logic in one place (the script)
- Supports per-repo additions via `.detect-secrets-ignore`
- Is consistent with `cloud-tools`, `en-ops-utils`, and other PDS repos

## 🩺 Additional context

See also the sister ticket in `template-repo-java` for the same update.

### PR DESCRIPTION
**Pull request by jordanpadams**
## 🗒️ Summary

Replaces the inline `slim-detect-secrets` pre-commit hook and duplicated `--exclude-files` argument lists with a centralized `scripts/detect_secrets_baseline.sh` script. Per-repo file exclusions now live in `.detect-secrets-ignore` (one regex per line) rather than being copy-pasted into every hook and workflow.

**Files changed:**
- `scripts/detect_secrets_baseline.sh` — new script (scan / audit / check modes)
- `.detect-secrets-ignore` — new per-repo exclusion file
- `.pre-commit-config.yaml` — replaced remote `slim-detect-secrets` hook with local script hook
- `.github/workflows/secrets-detection.yaml` — simplified to `pip install detect-secrets~=1.5.0` + run script; fails fast if `.secrets.baseline` is missing
- `setup.cfg` — added `detect-secrets~=1.5.0` to `[options.extras_require] dev`
- `README.md` — updated detect-secrets section to use script commands
- `CLAUDE.md` — added secrets detection section
- `docs/wiki-detect-secrets.md` — replacement content for the NASA-PDS wiki detect-secrets section

### 🤖 AI Assistance Disclosure
- [ ] No AI assistance used
- [ ] AI used for light assistance (e.g., suggestions, refactoring, documentation help, minor edits)
- [ ] AI used for moderate content generation (AI generated some code or logic, but the developer authored or heavily revised the majority)
- [x] AI generated substantial portions of this code

Estimated % of code influenced by AI: 90 %

## ⚙️ Test Data and/or Report

The script logic was validated against the existing pattern used in `cloud-tools` and `en-ops-utils`, which have been running this approach in production. The GitHub Actions workflow change is mechanical (same scan logic, delegated to script). No functional change to what is scanned.

## ♻️ Related Issues

Fixes #145

## 🤓 Reviewer Checklist

*Reviewers: Please verify the following before approving this pull request.*

### Documentation and PR Content
- [ ] **Documentation:** README, Wiki, or inline documentation (Sphinx, Javadoc, Docstrings) have been updated to reflect these changes.
- [ ] **Issue Traceability:** The PR is linked to a valid GitHub Issue
- [ ] **PR Title:** The PR title is "user-friendly" clearly identifying what is being fixed or the new feature being added, that if you saw it in the Release Notes for a tool, you would be able to get the gist of what was done.

### Security & Quality
- [ ] **SonarCloud:** Confirmed no new High or Critical security findings.
- [ ] **Secrets Detection:** Verified that the Secrets Detection scan passed and no sensitive information (keys, tokens, PII) is exposed.
- [ ] **Code Quality:** Code follows organization style guidelines and best practices for the specific language (e.g., PEP 8, Google Java Style).

### Testing & Validation
- [ ] **Test Accuracy:** Verified that test data is accurate, representative of real-world PDS4 scenarios, and sufficient for the logic being tested.
- [ ] **Coverage:** Automated tests cover new logic and edge cases.
- [ ] **Local Verification:** (If applicable) Successfully built and ran the changes in a local or staging environment.

### Maintenance
- [ ] **Backward Compatibility:** Confirmed that these changes do not break existing downstream dependencies or API contracts (or that breaking changes are clearly documented).

### CODE REVIEW

File: scripts/detect_secrets_baseline.sh [hunk 1]
```diff
@@ -0,0 +1,108 @@
...
+}
+
+if [ "$1" = "scan" ]; then
+    $DETECT_SECRETS scan "${EXCLUDE_ARGS[@]}" > .secrets.baseline
+    echo "Updated .secrets.baseline"
+    echo "Next step: run 'scripts/detect_secrets_baseline.sh audit' to review and classify detected secrets."
+elif [ "$1" = "audit" ]; then
+    $DETECT_SECRETS audit .secrets.baseline
+else
+    # Check 1: Fail if any secrets in the baseline have not been audited
```
Comment (nutjob4life) [inline scripts/detect_secrets_baseline.sh:79]: The CI version (GitHub Actions) has a nice explicit check for `.secrets.baseline` but this script falls through to Python to produce (possibly ugly) stack trace if it's missing. Should we have a similar check here?

File: scripts/detect_secrets_baseline.sh [hunk 2]
```diff
@@ -0,0 +1,108 @@
...
+")
+    if [ "$unaudited" -gt 0 ]; then
+        echo "⚠️ Attention Required! ⚠️" >&2
+        echo "$unaudited secret(s) in .secrets.baseline have not been audited." >&2
+        echo "Run 'scripts/detect_secrets_baseline.sh audit' to review and classify each detected secret." >&2
+        exit 1
+    fi
+
+    # Check 2: Fail if any new secrets are detected that are not in the baseline
+    cp .secrets.baseline .secrets.new
```
Comment (nutjob4life) [inline scripts/detect_secrets_baseline.sh:94]: Let's add an `EXIT` trap here so temporary files get cleaned up if anything gets interrupted:

    trap 'rm -f .secrets.new' EXIT

You can then remove (or keep if you like things explicit) all the other `rm -f .secrets.new`. (I vote keep, but I do get kind of explicit! 😎)

**Review by nutjob4life — APPROVED** [review]
Bravo! Thank you so much for buffing out the sharp "detect-secrets–shaped" edges in the template repository.

I heartily approve: ✅

**However**, I did notice a couple minor things in the script. Want to take a look? They could be ignored just fine, too.
